protocols/mdns/src/behaviour.rs

```diff
@@ -271,7 +273,7 @@ impl NetworkBehaviour for Mdns {
     }
 
     fn inject_new_listen_addr(&mut self, _id: ListenerId, _addr: &Multiaddr) {
-        self.send_buffer.push_back(build_query());
+        self.query = true;
```

Comment:
Instead of the extra `bool` why not do:

```suggestion
        self.timer = Timer::set_interval(Instant::now(), self.query_interval);
```

That would make the interval fire at the next `poll_next`. As far as I can tell there would be no need for `self.query` at all, right?